spec/jobs/movements_on_date_job_spec.rb

```diff
@@ -1,38 +1,76 @@
 require 'rails_helper'
 
 RSpec.describe MovementsOnDateJob, type: :job do
-  let!(:prison) { create(:prison, code: 'MDI') }
-  let(:nomis_offender_id) { 'G3462VT' }
-  let!(:alloc) { create(:allocation_history, nomis_offender_id: nomis_offender_id, secondary_pom_nomis_id: 123_435, prison: 'MDI') }
+  let(:nomis_offender_id) { nomis_offender.fetch(:offenderNo) }
+  let(:nomis_offender) { build(:nomis_offender, agencyId: to_prison_code) }
+  let(:case_info) { create(:case_information) }
+  let!(:offender_record) { create(:offender, nomis_offender_id: nomis_offender_id, case_information: case_info) }
+  let(:today) { Time.zone.today }
+  let(:yesterday) { today - 1.day }
+  let(:pom) { build(:pom) }
+  let!(:allocation) { create(:allocation_history, nomis_offender_id: nomis_offender_id, primary_pom_nomis_id: pom.staff_id, prison: from_prison_code) }
+  let(:mail_params) {
+    {
+      email: pom.email_address,
+      pom_name: pom.first_name.titleize,
+      offender_name: "#{nomis_offender.fetch(:lastName)}, #{nomis_offender.fetch(:firstName)}".titleize,
+      nomis_offender_id: nomis_offender_id,
+      prison_name: Prison.find(from_prison_code).name,
+      url: "http://localhost:3000/prisons/#{from_prison_code}/staff/#{pom.staff_id}/caseload"
+    }
+  }
 
   before do
-    stub_auth_token
-  end
+    stub_offender(nomis_offender)
+    stub_poms(from_prison_code, [pom])
+
+    # Stub offender movements from yesterday
+    stub_request(:get, "#{ApiHelper::T3}/movements?fromDateTime=#{yesterday - 1.year}T00:00&movementDate=#{yesterday}").
+      to_return(body: [movement].to_json)
 
-  it 'deallocates' do
-    allow(OffenderService).to receive(:get_offender).and_return(build(:hmpps_api_offender))
+    # Expect POM to be emailed about the deallocation
+    expect(PomMailer).to receive(:offender_deallocated).with(mail_params).and_return(double(deliver_later: nil))
```

Comment:
So this is stubbing at a lower level? - so just the movements? And then testing the PomMailer is able to receive a deallocated offender - with no Prison associated with it. That makes sense. 

Does it mater that it is testing PomMailer inside the MovementsOnDataJob spec?